File: Python/metrics/elementwise.py
import numpy as np

def ae(actual, predicted):
    return np.abs(np.array(actual)-np.array(predicted))

def mae(actual, predicted):
    return np.mean(ae(actual, predicted))

def mse(actual, predicted):
    return np.mean(se(actual, predicted))

def msle(actual, predicted):
    return np.mean(sle(actual, predicted))

def rmse(actual, predicted):
    return np.sqrt(mse(actual, predicted))

def rmsle(actual, predicted):
    return np.sqrt(msle(actual, predicted))

def se(actual, predicted):
    return np.power(np.array(actual)-np.array(predicted), 2)

def sle(actual, predicted):
    return (np.power(np.log(np.array(actual)+1) - 
            np.log(np.array(predicted)+1), 2))

File: C#/Metrics/Elementwise.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Koalas;

namespace Metrics
{
    public abstract class MeanError : Metric {
        public override bool IsMax { get { throw new NotImplementedException(); } }
        public override string Name { get { throw new NotImplementedException(); } }
        public override string Description { get { throw new NotImplementedException(); } }

        public override double Score(DataFrame solution, DataFrame submission)
        {
            var actual = (Series<double>) solution[-1];
            var predicted = (Series<double>) submission[-1];
            return actual.Zip(predicted, CellError).ToSeries().Mean();
        }

        public abstract double CellError(double actual, double predicted);
    }

    [Metric(Abbreviation = "MSE", Name = "Mean Squared Error")]
    public class MeanSquaredError : MeanError {
        public override double CellError(double actual, double predicted) {
            return Math.Pow(actual - predicted, 2.0);
        }
    }

    [Metric(Abbreviation = "MAE", Name = "Mean Absolute Error")]
    public class MeanAbsoluteError : MeanError {
        public override double CellError(double actual, double predicted) {
            return Math.Abs(actual - predicted);
        }
    }
}
